src/main/java/ru/practicum/shareit/user/UserController.java

```diff
@@ -1,12 +1,50 @@
 package ru.practicum.shareit.user;
 
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ru.practicum.shareit.user.dto.UserDto;
+import ru.practicum.shareit.user.model.User;
+
+import javax.validation.Valid;
+import java.util.List;
 
-/**
- * TODO Sprint add-controllers.
- */
 @RestController
+@Slf4j
+@Validated
+@RequiredArgsConstructor
 @RequestMapping(path = "/users")
 public class UserController {
+    private final UserService userService;
+
+    @PostMapping()
+    public UserDto createUser(@RequestBody @Valid User user) {
+        log.debug("Получен POST запрос на создание пользователя");
+        return userService.createUser(user);
+    }
+
+    @PatchMapping("/{userId}")
+    public UserDto updateUser(@PathVariable Long userId, @RequestBody User user) {
+        log.debug("Получен PATCH запрос на обновление пользователя пользователя с id: " + userId);
+        return userService.updateUser(userId, user);
+    }
+
+    @GetMapping
+    public List<UserDto> findAllUsers() {
+        log.debug("Получен GET запрос на получение списка всех пользователей.");
+        return userService.findAll();
+    }
+
+    @GetMapping("/{userId}")
+    public UserDto findUser(@PathVariable Long userId) {
```

Comment:
Падает, если пользователя нет
 GET http://localhost:8080/users/1

java.lang.NullPointerException: Cannot invoke "ru.practicum.shareit.user.model.User.getId()" because "user" is null
	at ru.practicum.shareit.user.UserMapper.toUserDto(UserMapper.java:12) ~[classes/:na]
	at ru.practicum.shareit.user.UserDaoImpl.getUser(UserDaoImpl.java:56) ~[classes/:na]
	at ru.practicum.shareit.user.UserServiceImpl.getUser(UserServiceImpl.java:33) ~[classes/:na]
	at ru.practicum.shareit.user.UserController.findUser(UserController.java:42) ~[classes/:na]